### PR TITLE
1947 - Convert ids-list-view test to playwright 

### LINKED ISSUE #1947: Convert List View Tests
Describe the task
Convert List View tests from https://github.com/infor-design/enterprise-wc/tree/main/tests-to-convert/ids-list-view to https://github.com/infor-design/enterprise-wc/blob/main/tests/ids-list-view/ids-list-view.spec.ts

Additional context
N/A

### PR DESCRIPTION
**Pull request by jbrcna**
**Explain the details for making this change. What existing problem does the pull request solve?**

**Related github/jira issue (required)**:

closes #1947 

**Steps necessary to review your pull request (required)**:

Run the test suite
Check the code conversion
New converted tests should pass

**Included in this Pull Request**:
- [ ] Some documentation for the feature.
- [ ] A test for the bug or feature.
- [ ] A note to the change log.

**Note:**
Test with TODO comments have errors.
The below functions does not exist:
activatedItem
Selected
getAllSwappableItems()
Have issues in childshlot & createFromTemplate(listView, html, container);

### CODE REVIEW

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 1]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+      { id: '1', subject: 'grape apple kiwi' },
+      { id: '2', subject: 'orange strawberry banana' },
+      { id: '3', subject: 'blueberry apricot' },
+      { id: '4', subject: 'pear banana raspberry' },
+      { id: '5', subject: 'apple blackberry starfruit' },
+      { id: '6', subject: 'kiwi orange apple' },
+      { id: '7', subject: 'blackberry raspberry blueberry' }
+    ];
+
+    test('renders the template without virtual scroll', async ({ page }) => {
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:207]: Remove `{ page }` as it was never used inside the test to fix lint errors. Apply to all test which has the same errors.

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 2]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+      await expect(idsListView).toHaveAttribute('item-height', '40px');
+      await idsListView.evaluate((listView: IdsListView) => { listView.itemHeight = undefined; });
+      await expect(idsListView).not.toHaveAttribute('height');
+    });
+
+    test('supports setting sortable', async ({ page }) => {
+      // TODO cannot set sortable
+      await idsListView.evaluate((listView: IdsListView) => { listView.sortable = true; });
+      await expect(idsListView).toHaveAttribute('sortable', 'true');
+      const swappableItem = idsListView.evaluate((listView: IdsListView) => listView.getAllSwappableItems()?.length);
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:293]: `getSwappableItems` is not an existing within IdsListView. Might want to try `itemsSwappable`.

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 3]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+    });
+
+    test('can single select with keyboard', async ({ page }) => {
+      // BUG?? cannot use keboard events.
+      // BUG?? Property 'selected' does not exist on type 'IdsListView'
+      let selected;
+      const sel = (nth: number) => `ids-list-view-item[row-index="${nth}"]`;
+
+      await idsListView.evaluate((listView: IdsListView) => { listView.selectable = 'single'; });
+      selected = await idsListView.evaluate((listView: IdsListView) => listView.selected);
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:350]: `selected` does not exist within the IdsListView. There's an `itemsSelected` that returns items selected in the IdsListView which you can check the length or use the IdsListViewItem to get if the item is `selected`.

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 4]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+    });
+
+    test('can multiple select with keyboard', async ({ page }) => {
+      // TODO
+      // BUG?? Property 'selected' does not exist on type 'IdsListView'
+      let selected;
+      const sel = (nth: number) => `ids-list-view-item[row-index="${nth}"]`;
+
+      await idsListView.evaluate((listView: IdsListView) => { listView.selectable = 'multiple'; });
+      selected = await idsListView.evaluate((lView: IdsListView) => lView.selected.length as IdsListViewSelectedItem[]);
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:389]: Same case as the previous `selected` comment. Property `selected` does not exists within the IdsListView object. Use `itemsSelected` if you wanted to validate multiple selected items.

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 5]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+      const sel = (nth: number) => `ids-list-view-item[row-index="${nth}"]`;
+      await page.evaluate(() => {
+        const elem: any = document.createElement('ids-list-view') as IdsListView;
+
+        elem.addEventListener('beforedeactivated', ((e: CustomEvent) => {
+          e.detail.response(veto);
+        }) as EventListener);
+      });
+      await page.click(sel(3));
+      activated = await idsListView.evaluate((lView: IdsListView) => lView.activatedItem); // activatedItem does not exist
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:618]: `activatedItem` is not existing within IdsListView. Try `itemsActivated` instead.

File: tests/ids-list-view/ids-list-view.spec.ts [hunk 6]
```diff
@@ -155,4 +175,1076 @@ test.describe('IdsListView tests', () => {
...
+      await expect(lvitems).toEqual(1);
+
+      await idsListView.evaluate((listview: IdsListView) => { (listview.searchField as any).value = ''; });
+      await expect(await idsListView.locator('ids-search-field input')).toHaveValue('');
+      lvitems = await idsListView.evaluate((listview: IdsListView) => listview.items.length);
+      await expect(lvitems).toEqual(itemCountAll);
+    });
+
+    test('fires filtered event when apply or clear search', async ({ page }) => {
+      // TODO how to use mockcallback??
```
Comment (n-ace-ancog) [inline tests/ids-list-view/ids-list-view.spec.ts:1204]: You can use the code below to check if the `filtered` event is being triggered in several scenarios.

```typescript
// Setup event listener - call only once
await idsListView.evaluate((element: IdsListView) => {
  (window as any).isEventTriggered = false;
  element.addEventListener('filtered', () => { (window as any).isEventTriggered = true; });
});

// Get value of isEventTriggered - re-use every scenario
const isEventTriggered = async () => {
  await page.evaluate(() => (window as any).isEventTriggered);
};

// Resets value of isEventTriggered - re-use every scenario
const resetEventTrigFlag = async () => {
  await page.evaluate(() => { (window as any).isEventTriggered = false; });
};
```

**Comment by jbrcna** [comment]
thanks for the comments @n-ace-ancog . will try to add those to my fix :thumbsup:

File: tests/ids-list-view/ids-list-view.spec.ts (hunk 5 above)
Comment (jbrcna) [reply tests/ids-list-view/ids-list-view.spec.ts]: in some cases itemsActivated has no call signatures

**Comment by jbrcna** [comment]
I need help on the ff:
ItemsActivated has no call signatures
Incorrect listview?.children?.length
Missing file extension for "../helpers/create-from-template"
child slot?

**Comment by tmcconechy** [comment]
@jbrcna i think we can pick those fixes up later. Just skip those tests for now?

**Review by n-ace-ancog — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]
Looks good to me!

**Pull request merged**